unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/MapRenderer/MinimapMetadata.cs

```diff
@@ -81,14 +66,38 @@ public void Clear()
         sceneInfoMap.Clear();
     }
 
-    private static MinimapMetadata minimapMetadata;
     public static MinimapMetadata GetMetadata()
     {
         if (minimapMetadata == null)
-        {
             minimapMetadata = Resources.Load<MinimapMetadata>("ScriptableObjects/MinimapMetadata");
-        }
 
         return minimapMetadata;
     }
-}
\ No newline at end of file
+
+    [Serializable]
+    public class MinimapSceneInfo
+    {
+        public string name;
+        public TileType type;
+        public List<Vector2Int> parcels;
+
+        public bool isPOI;
+        public string owner;
+        public string description;
+        public string previewImageUrl;
+
+        [NonSerialized]
+        private int hashCode = -1;
+
+        public override bool Equals(object obj) =>
+            obj != null && obj.GetHashCode() == GetHashCode();
+
+        public override int GetHashCode()
+        {
+            if (hashCode == -1)
+                hashCode = (name + type + parcels + isPOI + owner + description + previewImageUrl).GetHashCode();
```

Comment:
Prevent allocation, and include the first coordinate in the list, if available.
```suggestion
                hashCode = name.GetHashCode() + type.GetHashCode()  + parcels.GetHashCode()  + isPOI.GetHashCode()  + owner.GetHashCode()  + description.GetHashCode()  + previewImageUrl.GetHashCode() + parcels.Length + ((parcels.Length() > 0 ? parcels[0].X + parcels[0].Y * 600 : 0);
```